[12.0][MIG] account_bank_statement_import_mt940_base

feketemihai opened the pull request.

Commit:
[ENH] Support zip files.

sbidoul committed:
[IMP] rename bank_statement_parse_ to account_bank_statement_import_

Move parserlib to account_bank_statement_import and remove bank_statement_parse module

sbidoul committed:
[FIX] pep8

Commit:
[FIX] Import exceptions.Warning as UserError.

This according to OCA guidelines.
Also reformat README.rst to keep lines within limit.

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

hbrunn committed:
[UPD] parse more fields as of (#71)

* [UPD] parse more fields for MT940

as of https://www.ing.nl/media/ING_ming_mt940s_24_juli_tcm162-46356.pdf
p 8ff

* [FIX] There is no CR or LF in mt940 data to be parsed.

The calling logic from the MT940 parser concatenates all lines from a mt940 tag.

astirpe committed:
Extract method to allow extensions (#109)

luismontalba committed:
[10.0][NEW] account_bank_statement_import_move_line

oca-transbot committed:
OCA Transbot updated translations from Transifex

astirpe committed:
Porting to V10

astirpe committed:
Add tests

Commit:
[11.0] [MIG] account_bank_statement_import_mt940_base.

Update version to 11.

Get balance start even if currency code is already set.

Add SNS test, add header_lines parameter in parse file.

Update code, add pre process datafile to change from different format to only one, statements as dict starting with {4:..}, split file to import multiple statements at once, the check of date of statement import not implemented, be carefully not to import twice the same statements.

Update flake.

Add wrong file.

oca-travis committed:
[UPD] Update account_bank_statement_import_mt940_base.pot

Commit:
[MIG] account_bank_statement_import_move_line: Migration to 11.0

OCA-git-bot committed:
[ADD] icon.png

Commit:
[MIG] account_bank_statement_import_move_line: Migration to v12

Commit:
[12.0][IMP] use addres when no partner

when no partner_name send in bank statement address can be used
for partner identification

feketemihai committed:
Update contributor and module version.

wytzejan commented:
@feketemihai when do you think this PR is ready? I'm eagerly waiting for this one :)

feketemihai commented:
@wytzejan If i will have time, i will recheck tomorrow, since i think only some small issues with unit test.

pedrobaeza committed:
[MIG] account_bank_statement_import_move_line: Fix tests + URL

oca-travis committed:
[UPD] Update account_bank_statement_import_move_line.pot

OCA-git-bot committed:
[ADD] setup.py

alexey-pelykh committed:
[ADD] account_bank_statement_import_split: split statements during import

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (10 of 10 strings)

Translation: bank-statement-import-12.0/bank-statement-import-12.0-account_bank_statement_import_camt_oca
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-12-0/bank-statement-import-12-0-account_bank_statement_import_camt_oca/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 28.1% (9 of 32 strings)

Translation: bank-statement-import-12.0/bank-statement-import-12.0-account_bank_statement_import_move_line
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-12-0/bank-statement-import-12-0-account_bank_statement_import_move_line/hr/

oca-travis committed:
[UPD] Update account_bank_statement_import_split.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-statement-import-12.0/bank-statement-import-12.0-account_bank_statement_import_move_line
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-12-0/bank-statement-import-12-0-account_bank_statement_import_move_line/

OCA-git-bot committed:
[ADD] .editorconfig [ci skip]

bealdav committed:
IMP acc_bank_stat_imp_ofx: add checknum in name when required

oca-travis committed:
[UPD] Update account_bank_statement_import_ofx.pot

OCA-git-bot committed:
account_bank_statement_import_ofx 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-statement-import-12.0/bank-statement-import-12.0-account_bank_statement_import_ofx
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-12-0/bank-statement-import-12-0-account_bank_statement_import_ofx/

OCA-git-bot committed:
account_bank_statement_import_camt_oca 12.0.1.0.1

feketemihai committed:
Restructure module for MT940 import, use RABO as general implementation.

Fix pylint.

Update pylint line error

feketemihai commented:
@sbidoul @hbrunn @astirpe Can you check this one, please. I restructure the module and use RABO as a general implementation, for other formats, we should inherit the methods to give the right returns, added methods for get_header_lines, get_header_regex, get_footer_regex, get_codewords, get_tag_61_regex... to be more easy. I will try to migrate the ING one for example.

feketemihai committed:
Improve unit test.

Improve unit test.

Improve unit test.

Improve unit test.

feketemihai committed:
Merge branch '12.0' of github.com:OCA/bank-statement-import into 12.0-mig-account_bank_statement_import_mt940_base

feketemihai committed:
Remove unit test tags.

feketemihai committed:
Remove unit test tags.

feketemihai committed:
Update module MT940 base

hbrunn commented:
for my customers, MT940 is pretty much obsolete (they all use CAMT now), so I probably won't be of much help here

pedrocs-exo commented:
Hi, will this PR ever be ready and approved? The code works and it's still useful in some cases. How can I help?

pedrocs-exo commented on account_bank_statement_import_mt940_base/models/mt940.py:
```diff
@@ -0,0 +1,359 @@
+            header_lines = self.get_header_lines()
+        for match in matches:
+            self.is_mt940_statement(line=match)
+            iterator = '\n'.join(
+                match.split('\n')[1:-1]).split('\n').__iter__()
```
Should be:

```
            iterator = '\n'.join(
                match.split('\n')[1:]).split('\n').__iter__()
```

instead. Otherwise the last line will be ignored. One of the banks I'm working on uses it so I guess there could be more.
